app/src/main/java/com/android/bookswap/model/chat/MyFirebaseMessagingService.kt

```diff
@@ -0,0 +1,118 @@
+package com.android.bookswap.model.chat
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat.getSystemService
+import com.android.bookswap.MainActivity
+import com.android.bookswap.R
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+/** Service to handle Firebase Cloud Messaging notifications. */
+const val DEFAULT_TITLE = "Hello There!"
+const val DEFAULT_BODY = "You have a new notification"
+
+open class MyFirebaseMessagingService : FirebaseMessagingService() {
+  /**
+   * Called when a message is received.
+   *
+   * @param remoteMessage the message received from Firebase Cloud Messaging.
+   */
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    val notification = remoteMessage.notification
+    val title = notification?.title ?: DEFAULT_TITLE
+    val body = notification?.body ?: DEFAULT_BODY
+    sendNotification(title, body)
+  }
+
+  /**
+   * Sends a notification to the user.
+   *
+   * @param title the title of the notification.
+   * @param messageBody the body of the notification.
+   */
+  fun sendNotification(title: String, messageBody: String) {
+    // Create an intent to open the MainActivity when the notification is clicked
+    val intent =
+        Intent(this, MainActivity::class.java).apply { addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP) }
+    val pendingIntent =
+        PendingIntent.getActivity(
+            this, 0, intent, PendingIntent.FLAG_ONE_SHOT or PendingIntent.FLAG_IMMUTABLE)
+    // Create intent and PendingIntent for the "Accept" action
+    val acceptIntent =
+        Intent(this, NotificationActionReceiver::class.java).apply { action = "ACTION_ACCEPT" }
+    val acceptPendingIntent =
+        PendingIntent.getBroadcast(
+            this,
+            0,
+            acceptIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+
+    // Create intent and PendingIntent for the "Decline" action
+    val declineIntent =
+        Intent(this, NotificationActionReceiver::class.java).apply { action = "ACTION_DECLINE" }
+    val declinePendingIntent =
+        PendingIntent.getBroadcast(
+            this,
+            0,
```

Comment:
Both the "Accept" and "Decline" actions use the same request code (0) for creating the PendingIntent. This could cause unexpected behavior since they are distinct actions but share the same request code.

Reply:
yes it 's fix in  an  another PR , but thanks you for the suggestion